### ISSUE TITLE
Multiple customs domains?

### ISSUE THREAD

**Issue by kevinrenskers** [sintaxi/surge#79]
A huge problem with GitHub pages for me is that they only support one custom domain per site. Since my site has multiple domains (.nl and .com tld's) it would be so nice to have support for multiple domains with Surge. 

Also consider stuff like mydomain.com, my-domain.com, mydomain.net and my-domain.net etc etc. Quite common to have multiple domains for the same site. Currently I'm stuck with making redirecting index.html pages for the other domains since my DNS registrar doesn't offer DNS directs.

I know I could find another registrar where I could redirect my other domains to my main domain, but native Surge support for multiple domains seems like a huge +1 over GitHub :)

**Comment by silentrob** [comment]
Just to be clear, we are talking about redirecting other domains to a single project, and not publishing one project to many domains. That would be bad for SEO.

**Comment by kevinrenskers** [comment]
Yep. I would imagine that the CNAME file could have multiple lines, the first line is the main domain. Something like that.

**Comment by silentrob** [comment]
Would you consider this a pro feature or just expect it to work in the free tier?

**Comment by kevinrenskers** [comment]
Depends on the price of the pro tier.. but I'd pay for it.

**Comment by gyaresu** [comment]
I haven't seen anything on pro features so I'm guessing that's to come.

I just had the exact desire to push a couple of "doing nothing" domains to a single surge instance.

I even tried a multi-line CNAME file to see if it would work because that seemed most logical.

The question then becomes about your system resources vs circumvention.

If all one has to do is alias a push to multiple surge instances then perhaps it's better for you to do that in DNS?

 * `alias live =  'surge -d example.com /path ; surge -d example2.com /path;'`

**Comment by kennethormandy** [comment]
Hey, just an update on this. We don’t have any short-term plans to add support for listing multiple domains in the `CNAME` file, but this probably would be a paid feature if/when we were to add it.

In the meantime, you can achieve the same result through one of these two options.

1. Publish the same site to multiple domains, and alias it to some kind of deployment task
2. Publish the first site to the primary domain, and redirect to it in any subsequent domains using a `ROUTER` file.

The first, which @gyaresu mentioned, is to publish the same project to multiple domain names, and alias the deployment to a [Grunt task](https://github.com/surge-sh/grunt-surge), Gulp taks, or npm run script, like in this `package.json` file:

```js
"devDependencies": {
  "surge": "latest"
},
"scripts": {
  "deploy": "surge ./path/to/dist example.com; surge ./path/to/dist example2.com"
}
```

Alternatively, you can publish to the primary domain as normal add add a redirect to it into by publishing with a `ROUTER` file in your second project with your second domain. This is a paid feature, and is more SEO-friendly. The primary domain is used for the real site, and the second domain gets a permanent redirect to the primary one. To add a redirect for your other domain, add [a `ROUTER` file](https://surge.sh/help/adding-redirects) into the root of the project.

The `ROUTER` file might look like this:

```
301   /         http://example.com
```

You can even add multiple redirects so more paths end up at the proper domain:

```
301   /         http://example.com
301   /:title   http://example.com/:title
```

Then, you would publish it to `example2.com`:

```sh
surge ./path/to/dist example2.com
```

Either way, the site will now be available through both your domains. Let me know if you have any feedback on either approach!

**Comment by mihalik** [comment]
I'm interested in pointing multiple subdomains of a wildcard domain to the same project.  Similar to how Slack uses surge-sh.slack.com and project.slack.com to differentiate clients and allow users to be logged into multiple accounts at once.  The code I'm deploying is a SPA that will read the domain and cookies to handle differentiate clients.

Using the default implementation I could deploy the same project to multiple subdomains but I'd like to avoid having to specify the subdomains and I'd like to avoid the Surge.sh "Project Not Found" page for unknown clients.

Ideally I could do something like this:

* www and naked domain would point to one project
* wildcard would point to another project
